operator/src/lib.rs

```diff
@@ -266,7 +190,7 @@ impl SparkState {
 
                 trace!("{}", message);
 
-                self.status = self
+                self.context.resource.status = self
```

Comment:
I know this isn't part of this PR, but I think the hardcoded `ConditionStatus::True` below is wrong.